config_v2.md

```diff
@@ -0,0 +1,225 @@
+# Sacred config v2
+
+## Goals of this document:
+
+* Present the API for the new way to work with sacred configs
+
+* Make sure this API can handle all the use case that used to work before.
+
+* Make sure that this API is intuitive and Pythonic.
+
+* Make sure that someone who has never read the sacred docs can understand some codebase that uses sacred.
+
+* Remove all black magic of configs. It is linked to the previous point.
+
+* Give enough flexibility to allow implementation of many of the features requests about configs in github issues.
+
+## Non-goals of this document:
+
+* Describing or even caring about the migration process. It can be done at a later stage.
+
+* Caring about looking like our old API about configs. Legacy API design should not prevent us from pushing a better API.
+
+* Handling seeding. The seeding has been separated from the configuration process with the PR [Making the seeding more intuitive](https://github.com/IDSIA/sacred/pull/615)
+
+
+## API description:
+
+The goal is to make the API easier to understand.
+
+* There is no more `Ingredient`. 
+
+* There is no config scopes.
+
+* Ingredients are replaced by functions which can be activated either from the command line or when creating the Experiment.
```

Comment:
I don't understand this suggestion. I mainly use Ingredient's to re-use config options for some common sub-component. This is nice as it pulls in the default config and all named configs. It seems hard to replicate that just by calling config functions manually. Could you elaborate on this?

Reply:
Sure it's also linked to my comment here: https://github.com/IDSIA/sacred/pull/623#discussion_r321965014 we should be able to add other potential modifications from one potential modification. Let me update the document with a new section with more details.

Reply:
I've added a section about default config and pulling named configs. Please take a look.